### PR TITLE
basic webui for log search

### PR DESCRIPTION
**Pull request by amanjiofficial**
Hey @Ali-Razmjoo. I have tried to implement the search UI. Please review the code and suggest additional changes to enhance this feature.
Issue : #67

### CODE REVIEW

**Comment by amanjiofficial** [comment]
@dhirensr any suggestions on this implementation. 
Regarding export of logs should we save the file that is being exported for our reference or just allow the user to download the file.

File: api/server.py [hunk 1]
```diff
@@ -1044,6 +1044,41 @@ def top_usernames_used():
...
+            )
+        },
+        {
+            "$limit": fix_limit(
+                get_value_from_request("limit")
+            )
+        }
+    ]
+    if module_name:
+        module_query.insert(0, {"$match": {"$or": [ {'username': search_val, 'module_name': module_name}, {'ip_dest': search_val, 'module_name': module_name} ] } })
```
Comment (dhirensr) [inline api/server.py]: indentation problem 
please not more than 80 chars in a line

File: api/server.py [hunk 2]
```diff
@@ -1044,6 +1044,41 @@ def top_usernames_used():
...
+    else:
+        module_query.insert(0, {"$match": {"$or": [ {'username': search_val}, {'ip_dest': search_val} ] } }
```
Comment (dhirensr) [inline api/server.py]: bracket missing problem.
please test the code first and then send the PR.

File: api/server.py [hunk 3]
```diff
@@ -1075,6 +1110,40 @@ def top_passwords_used():
...
+            "$limit": fix_limit(
+                get_value_from_request("limit")
+            )
+        }
+    ]
+    if module_name:
+        module_query.insert(0, {"$match": {"$or": [ {'password': search_val, 'module_name': module_name}, {'ip_dest': search_val, 'module_name': module_name} ] } })
+    else:
+        module_query.insert(0, {"$match": {"$or": [ {'password': search_val}, {'ip_dest': search_val} ] } })
+        try:
```
Comment (dhirensr) [inline api/server.py]: indentation problem

File: api/server.py (hunk 2 above)
Comment (amanjiofficial) [reply api/server.py]: It must have been done unintentionally in the second commit( for removing whites paces) that I later squashed to maintain the branch. Point noted and will be taken care.

**Comment by dhirensr** [comment]
@amanjiofficial : could you also the tests for the route in the new file named test_api.py (please make it in directory https://github.com/zdresearch/OWASP-Honeypot/tree/master/tests) ? so I would review it all together then :smile:

**Comment by dhirensr** [comment]
@amanjiofficial : 
somewhere there is a bug in the server I feel because when I checked  for 
http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=%22%22&password=1
and 
http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=%22%22&password=1
the answers are different and the same is happening in the UI.

**Comment by amanjiofficial** [comment]
Though I have handled "All Modules" case in a similar way as done in already implemented route "most-usernames-used", yet I will check once again and implement the changes if required.@dhirensr

**Comment by amanjiofficial** [comment]
> @amanjiofficial :
> somewhere there is a bug in the server I feel because when I checked for
> http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=&password=1
> and
> [http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=""&password=1](http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=%22%22&password=1)
> the answers are different and the same is happening in the UI.

A get request for both url returns same result.
Link: `https://imgur.com/a/IVglyVs`.

Is there any specific configuration to be followed for database for test of this route as that could be further used in testing all the routes.
@dhirensr @Ali-Razmjoo

**Comment by dhirensr** [comment]
@Ali-Razmjoo : waiting for your review on this

**Comment by ChakshuGupta** [comment]
@amanjiofficial 
When both module_name and search_val are empty, it should return all the records. It is returning empty string.
This also causes an error when searching with the UI as when you remove your search parameter, the table becomes empty. It should fill up with all the values again.

http://127.0.0.1:5000/api/events/most-passwords-used-search?module_name=&search_val=%22%22

http://127.0.0.1:5000/api/events/most-usernames-used-search?module_name=&search_val=%22%22

**Comment by dhirensr** [comment]
closing this for now because it's inactive since 2months

**Pull request closed without merging**